pkg/vpcmodel/connectivityAbstraction.go

```diff
@@ -70,68 +86,20 @@ func partitionConnectivityByNodeSet(nodesConn GeneralConnectivityMap, nodeSet No
 	return otherToOther, nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet
 }
 
-// checkConnectivityAbstractionValidity() checks if the abstraction assumption holds
-// it does it on each group separately.
-// for now it creates a string
-// todo: - how to report this string? what format?
-// for now, No need to CR it, we do not do anything with the result
-
-// func checkConnectivityAbstractionValidity(connMap GeneralConnectivityMap, nodeSet NodeSet, isIngress bool) {
-// 	res := ""
-// 	for node1, nodeConns := range connMap {
-// 		allConns := map[string][]VPCResourceIntf{}
-// 		for node2, conn := range nodeConns {
-// 			allConns[conn.String()] = append(allConns[conn.String()], node2)
-// 		}
-// 		if len(allConns) > 1 || len(nodeConns) != len(nodeSet.Nodes()) {
-// 			directionAdjective := map[bool]string{false: "from", true: "to"}[isIngress]
-// 			res += fmt.Sprintf("node %s has different access %s %s:\n", node1.Name(), directionAdjective, nodeSet.Name())
-// 			if len(nodeConns) != len(nodeSet.Nodes()) {
-// 				res += fmt.Sprintf("    it has no access %s ", directionAdjective)
-// 				for _, node2 := range nodeSet.Nodes() {
-// 					if _, ok := nodeConns[node2]; !ok {
-// 						res += fmt.Sprintf("%s, ", node2.Name())
-// 					}
-// 				}
-// 				res += "\n"
-// 			}
-// 			for conn, nodes := range allConns {
-// 				res += "    "
-// 				if isIngress {
-// 					res += fmt.Sprintf("%s -> ", node1.Name())
-// 				}
-// 				for _, n := range nodes {
-// 					res += fmt.Sprintf("%s,", n.Name())
-// 				}
-// 				if !isIngress {
-// 					res += fmt.Sprintf(" -> %s", node1.Name())
-// 				}
-// 				res += fmt.Sprintf(" %s\n", conn)
-// 			}
-// 		}
-// 	}
-// }
-
-// mergeConnectivityWithNodeSetAbstraction() merge the four groups, while abstracting the connections
-func mergeConnectivityWithNodeSetAbstraction(
-	otherToOther, nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet GeneralConnectivityMap,
+// mergeConnectivityWithNodeSetAbstraction() merge the three last groups, while abstracting the connections
+func (nsa *NodeSetAbstraction) mergeConnectivityWithNodeSetAbstraction(
+	nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet GeneralConnectivityMap,
 	nodeSet NodeSet) GeneralConnectivityMap {
-	// first make a copy of otherToOther, to be the result:
-	res := GeneralConnectivityMap{}
-	for src, nodeConns := range otherToOther {
-		for dst, conns := range nodeConns {
-			res.updateAllowedConnsMap(src, dst, conns)
-		}
-	}
 	// all the connections with the nodeSet are merged to *only* one connectivity, which is the union of all separate connections:
+	mergedConnectivity := GeneralConnectivityMap{}
 	allConns := NoConns()
 	for _, nodeConns := range nodeSetToNodeSet {
 		for _, conns := range nodeConns {
```

Comment:
Consider adding a function for the inner loop and call it 3 time instead of duplicating it

Reply:
each loop is different, please notice,
we discussed it on abstraction PR

Reply:
I remember our discussion.
However, I was referring to the following code which is indeed duplicated 3 times: 
```
		for _, conns := range nodeConns {
			allConns = allConns.Union(conns)
		}
```

Reply:
fixed

Reply:
cool - I was not aware of this coding alternative :-)